fabric-input-events-v1/src/main/java/net/fabricmc/fabric/api/event/client/input/MouseMoveEvent.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.client.input;
+
+import net.fabricmc.fabric.api.client.input.v1.FabricMouse;
+
+public class MouseMoveEvent extends GenericMouseEvent {
```

Comment:
This is gonna allocate a whole new object every single time the mouse moves. Not acceptable for literally four doubles and two ints. Pass them directly to the lambda.

Reply:
See `GenericMouseEvent`, there is a lot more information that is relevant for each event. Also, see the comment above.